### DISCUSSION TITLE
Helper for resp.ToBytes()?

### DISCUSSION THREAD

**Discussion by 0x079** [redis/rueidis#180] [Ideas]
There are response helpers for pretty much everything except []byte.
We have to do 
```
str, err := resp.ToString()
bin := []byte(str)
```
or
```
data, err := resp.ToAny()
bin := data.([]byte)
```
but that's not very convenient and can add up to a lot across the codebase. It would be nice to have `resp.ToBytes()` unless I'm missing something?

**Comment by rueian** [comment]
Depending on your use case, `resp.AsReader()` might suit you better, because it returns an `io.Reader` which can be consumed by many decoders, such as `json.Decoder` and `hex.Decoder`.

Besides that, I do have the plan of adding `resp.AsBytes()` helper in go 1.20.

**Reply by 0x079** [reply]
That's awesome! Thank you!

**Reply by bRhoa** [reply]
Are you planning on allowing bytes in your `Key()` command builder instead of requiring `rueidis.BinaryString(bytes)` in 1.20?

**Reply by rueian** [reply]
Probably no. I believe that only accepting `string` as the input of `Key()` is the most user-friendly signature.

**Reply by rueian** [reply]
Hi @0x079, `resp.AsBytes()` has been introduced in the latest v0.0.96.
